src/Controllers/ProfileController.js

```diff
@@ -0,0 +1,147 @@
+import mongoose from "mongoose";
+import { userProfileModel } from "../Models/UserProfileModel.js";
+import { userModel } from "../Models/UserModel.js";
+import { CustomError } from "../Utils/CustomError.js";
+import * as service from "../service.js";
+
+/*
+User profile get request method to fetch the user profile contents from the token containing the user UID
+*/
+async function getProfile(req, res) {
+  try {
+    const authorizationData = service.getAuthorizationData(req);
+    if (!authorizationData?.userId) {
+      res.setHeader("WWW-Authenticate", 'Basic realm="user"');
+      throw new CustomError("Invalid authorization data", "AuthorizationError", 401);
+    }
+    try {
+      const user = await userProfileModel.findOne({ email: authorizationData.userId }).lean();
+      if (!user) {
+        throw new CustomError("No linked user profile found", "NotFoundError", 404);
+      } else {
+        res.status(200).json(user);
+      }
+    } catch (err) {
+      if (err instanceof CustomError) throw err;
+      // Handle gateway timeout error
+      res.status(504).send();
+    }
+  } catch (err) {
+    res.status(err.status_code).send({ Type: err.name, Error: err.message });
```

Comment:
*Transcribed from Discord for completeness:*

> Could you tell me a bit about the pattern that should be employed here?

Throwing errors disrupts program flow and should only be done when program flow *needs* to be disrupted.  `try-catch` blocks also serve as warnings to other developers of potential points of failure.

In this case, instead of throwing a `CustomError`, it's better to modify `res` appropriately & call `.send()`, then let program flow cascade out of the blocks normally.

The `try-catch` block around the call to `userProfileModel.findOneAndDelete()` is appropriate, though, because failing to communicate with the database *is* an exceptional condition.